storage/src/lib.rs

```diff
@@ -328,46 +329,44 @@ impl Storage {
         Ok(len)
     }
 
+    pub fn path(key: &str) -> String {
+        format!("{}{}", DATA_PATH, key)
+    }
+
     pub async fn put_json_slice<'a>(&self, key: &'a str, json: &'a [u8]) -> Result<usize, Error> {
         let mut stream = once(ok::<_, std::io::Error>(json));
         self.put_stream(key, "application/json", None, &mut stream).await
     }
 
-    // This will load the entire S3 object into memory
-    pub async fn get(&self, key: &str) -> Result<Vec<u8>, Error> {
-        let path = format!("{}{}", DATA_PATH, key);
-        self.get_object(&path).await
-    }
-
-    pub async fn get_head(&self, key: &str) -> Result<Head, Error> {
-        let path = format!("{}{}", DATA_PATH, key);
+    pub async fn get_head(&self, path: &str) -> Result<Head, Error> {
```

Comment:
I'm a bit worried about this change, because it can cause accidental/wrong use of the API. Instead, maybe we could do this:

```
pub struct Key {
  k: String
}

impl Key {
    // No prefix
    pub fn absolute(&str) -> Key {}
    // Relative to base
    pub fn relative(&str) -> Key {}
}
```

And then we can make the APIs take a &Key instead of a &str.

Reply:
Agree totally. I was playing around with it a bit, but didn't want to change too much before wider discussion. I'll try to implement these changes next.

Reply:
@lulf first attempt at improving the API [#4c2e73](https://github.com/trustification/trustification/pull/482/commits/4c2e73b8fcca13b8cff153ace10fe55229c20443)